src/libsystemd/sd-device/sd-device.c

```diff
@@ -1180,21 +1180,14 @@ static int device_set_sysname_and_sysnum(sd_device *device) {
                 return r;
 
         /* some devices have '!' in their name, change that to '/' */
-        while (sysname[len] != '\0') {
-                if (sysname[len] == '!')
-                        sysname[len] = '/';
-
-                len++;
-        }
-
-        /* trailing number */
-        while (len > 0 && isdigit(sysname[--len]))
-                sysnum = &sysname[len];
+        for (p = strchrnul(sysname, '!'); *p != '\0'; p = strchrnul(p, '!'))
+                *p = '/';
```

Comment:
hmm, why not use regular strchr()? sounds simpler:

```c
for (p = strchr(sysname, '!'); *p; p = strchr(p, '!'))
    *p = '/';
```

Reply:
btw, we have a very similar function efi_tilt_backslashes() already. maybe we should unify this into a common helper?

i.e. `char *string_replace_char(char *buf, char old_char, char new_char);`

and then make efi_tilt_backslashes() just a trivial inline wrapper around this?